functions/fish_prompt.fish

```diff
@@ -895,9 +895,38 @@ function __bobthefish_prompt_desk -S -d 'Display current desk environment'
     set_color normal
 end
 
+function __bobthefish_prompt_find_file_up -S -d 'Find file(s), going up the parent directories'
+    set -l dir "$argv[1]"
+    set -l files $argv[2..]
+
+    if [ -z "$dir" ]
+        return 1
+    end
+
+    while [ "$dir" ]
```

Comment:
won't this loop forever because `__bobthefish_dirname /` resolves to `/`?

[the mercurial equivalent to this](https://github.com/oh-my-fish/theme-bobthefish/blob/master/functions/fish_prompt.fish#L237-L238) has the root directory special cased.

Reply:
Interesting... and fixed.

<img width="613" alt="image" src="https://user-images.githubusercontent.com/681969/151853675-212de059-aa2f-46c1-9abc-8a5440fe437a.png">

Reply:
Also note there's a refactor we can do to use this new function instead of rolling it locally.